modules/carto/src/sources/utils.ts

```diff
@@ -0,0 +1,63 @@
+import {APIErrorContext, CartoAPIError} from '../api/carto-api-error';
+import {encodeParameter, MapType} from '../api/maps-api-common';
+import {DEFAULT_HEADERS, DEFAULT_PARAMETERS, MAX_GET_LENGTH} from './common';
+import {buildMapsUrlFromBase} from '../config';
+
+export async function requestWithParameters<T = any>({
+  baseUrl,
+  parameters,
+  headers: customHeaders,
+  errorContext
+}: {
+  baseUrl: string;
+  parameters?: Record<string, string>;
+  headers: Record<string, string>;
+  errorContext: APIErrorContext;
+}): Promise<T> {
+  let url = baseUrl;
+  if (parameters) {
+    const allParameters = {...DEFAULT_PARAMETERS, ...parameters};
+    const encodedParameters = Object.entries(allParameters).map(([key, value]) => {
+      return encodeParameter(key, value);
+    });
+    url += `?${encodedParameters.join('&')}`;
+  }
+
+  const headers = {...DEFAULT_HEADERS, ...customHeaders};
+  try {
+    /* global fetch */
+    let response: Response;
+    if (url.length > MAX_GET_LENGTH) {
+      response = await fetch(url, {method: 'POST', body: JSON.stringify(parameters), headers});
+    } else {
+      response = await fetch(url, {headers});
+    }
+    let json: any;
+    try {
+      json = await response.json();
+    } catch {
+      json = {error: ''};
+    }
+    if (!response.ok) {
+      throw new CartoAPIError(json.error, errorContext, response);
```

Comment:
@felixpalmer the error is always `{error: ''}`, do we have a way to know the error?

Reply:
This is how the current code works. `CartoAPIError` will display an error message which will log out the response error code. The error will only by `''` when no JSON response can be parsed

Reply:
Ok, thx!